src/admin/content-block/content-block.types.ts

```diff
@@ -1,7 +1,60 @@
-export enum ContentBlockType {
-	Heading = 'Heading',
-	RichText = 'RichText',
-	RichTextTwoCols = 'RichTextTwoCols',
+export type ContentBlockStateType = 'components' | 'block';
+
+export type ContentBlockStateOptions =
+	| Partial<ContentBlockComponentState>
+	| Partial<ContentBlockComponentState>[]
+	| Partial<ContentBlockState>;
+
+export type AlignOptions = 'left' | 'right' | 'center';
+
+export type HeadingLevelOptions = 'h1' | 'h2' | 'h3' | 'h4';
+
+// CONTENT BLOCK CONFIG
+export interface ContentBlockMeta {
+	index: number;
+	config: ContentBlockConfig;
+}
+
+export interface ContentBlockConfig {
+	name: string;
+	components: ContentBlockComponentsConfig;
+	block: ContentBlockBlockConfig;
+}
+
+export interface ContentBlockComponentsConfig {
+	name?: string;
+	state: ContentBlockComponentState | ContentBlockComponentState[];
+	fields: {
+		[key: string]: ContentBlockField;
+	};
+}
+
+export interface ContentBlockBlockConfig {
+	state: ContentBlockState;
+	fields: {
+		[key: string]: ContentBlockField;
+	};
+}
+
+export type ContentBlockComponentState =
+	| HeadingBlockComponentState
+	| RichTextBlockComponentState
+	| RichTextTwoColumnsBlockComponentState
+	| ButtonsBlockComponentState
+	| IntroBlockComponentState;
+
+export type ContentBlockState =
```

Comment:
Is er een usecase voor ContentBlockState ipv DefaultContentBlock te gebruiken?
De props van de wrapper block moeten toch altijd hetzelfde zijn en enkel de ContentBlockComponentState verschilt per block?
Indien we nog geen usecase hiervoor hebben dank ik dat het beter is dat we voorlopig DefaultContentBlock  gebruiken en dit uitbreiden op het moment dat er een usecase gevonden wordt die aparte ContentBlockStates vereist